File: golang/create/builder.go
package main
type ResourceConf struct {
    maxTotal int
    maxIdle int
}
func (r *ResourceConf) getTotal() int {
    return r.maxTotal
}
func (r *ResourceConf) getIdle() int {
    return r.maxIdle
}
type Builder struct {
    maxTotal int
    maxIdle int
}
func (b *Builder) build() *ResourceConf {
    return &ResourceConf{b.maxTotal, b.maxIdle}
}
func (b *Builder) setTotal(total int) *Builder {
    b.maxTotal = total
    return b
}
func (b *Builder) setIdle(idle int) *Builder {
    b.maxIdle = idle
    return b
}
func main() {
    b := &Builder{}
    r := b.setTotal(10).setIdle(2).build()
    println(r.getTotal(), r.getIdle())
}

File: cpp/create/builder.cpp
#include<iostream>
using namespace std;

class ResourceConf {
private:
    int maxTotal;
    int maxIdle;

public:
    ResourceConf(int total, int idle):maxTotal(total), maxIdle(idle) {}
    class Builder;
    int getMaxTotal() {
        return maxTotal;
    }
    int getMaxIdle() {
        return maxIdle;
    }
};

class ResourceConf::Builder {
private:
    int maxTotal;
    int maxIdle;
public:
    Builder* setTotal(int total) {
        maxTotal = total;
        return this;
    }
    Builder* setIdle(int idle) {
        maxIdle = idle;
        return this;
    }
    ResourceConf* build() {
        return new ResourceConf(maxTotal, maxIdle);
    }
};

int main() {
    ResourceConf& conf = *(new ResourceConf::Builder())->setTotal(10)->setIdle(2)->build();
    cout <<conf.getMaxTotal() <<":" <<conf.getMaxIdle() <<endl;
    return 0;
}